### DISCUSSION TITLE
Import symbols from modules

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#617] [Ideas & RFCs]
Can we improve the readability and efficiency of our code by importing commonly used functions from `ftd` at the top of a file and using them directly in the code to avoid repeating long function names like `ftd.row` throughout the code?

For example:
```ftd
-- import: row, col, img, text-input from ftd
```

Code:
```ftd
-- col:
align-content: center
width: fill-conainer

-- img:
src:

-- end: col
```

Asked by @saurabh-lohiya
(Rephrased)

**Comment by amitu** [comment]
This is an interesting suggestion, and we have considered it, something like:

```
;; if you want to import everything
-- import: ftd
exposing: *

;; or if you want to import only a few items
-- import: ftd
exposing: row, column
```

While this reduces typing, it increases cognitive load when reading. Every time I come across `row` I have to not go to the top of the file to check if it is indeed `ftd.row` or `row` defined in some other package. 

We believe it would be better to first give a better writing support using [`lsp`](https://github.com/fastn-stack/fastn/discussions/609), so editors can give auto complete etc, to facilitate writing.

**Comment by amitu** [comment]
This is already implemented, the following works:

```ftd
-- import: fastn-community.github.io/design-system/typography
export: heading-hero, heading-large, heading-medium
exposing: heading-large, copy-regular
```

`export: *` and `exposing: *` does not work yet, but the original ask can be satisfied with what we have.
